lib/puma/dsl.rb

```diff
@@ -494,8 +494,9 @@ def after_worker_fork(&block)
 
     # Code to run out-of-band when the worker is idle.
     # These hooks run immediately after a request has finished
-    # processing and there are no busy threads on the worker.
-    # The worker doesn't accept new requests until this code finishes.
```

Comment:
Whoops, that wasn't ever true, was it?

Reply:
Also, Q: should this be the actual behavior? Is it really OOB otherwise?

Reply:
> Whoops, that wasn't ever true, was it?

The oob hook previously ran from the acceptor thread, so I think it _was_ true at least for the case where it was working correctly (max threads 1).

> should this be the actual behavior? Is it really OOB otherwise?

Hmm, I went back and forth on whether it's better to keep accepting new requests while an OOB hook is running. (Might depend on the hook's use case?) For GC optimization at least, I thought continuing to accept requests could potentially be better under high load (though the MRI GVL could make it a wash). But on second thought, _not_ accepting new requests is probably safer for light loads especially while request-balancing is still uneven. Plus it is 'really OOB' for other use-cases. I can change this to run on the acceptor thread as before.

Reply:
Updated in 7b3639125d8591103db7962f88531208ca716aff